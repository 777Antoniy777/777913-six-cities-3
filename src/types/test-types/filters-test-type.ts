type TestFilter = {
  id: string,
  value: string,
};

type TestFilters = TestFilter[];

export {TestFilter, TestFilters};
