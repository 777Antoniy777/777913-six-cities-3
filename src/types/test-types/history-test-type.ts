type TestRouteHistory = {
  push: () => void;
};

export {TestRouteHistory};
