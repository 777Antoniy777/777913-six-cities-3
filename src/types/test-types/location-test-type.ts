type TestRouteLocation = {
  pathname: string;
};

export {TestRouteLocation};
