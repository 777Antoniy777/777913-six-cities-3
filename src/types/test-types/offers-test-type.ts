type TestCity = {
  id: number;
  name: string;
  coords: number[];
};

type TestHost = {
  avatar: string;
  name: string;
  status: boolean;
};

type TestReview = {
  id: number;
  body: string;
  rating: number;
  name: string;
  date: string;
};

type TestOffer = {
  id: number;
  city: TestCity;
  title: string;
  premium: boolean;
  src: string;
  photos: string[];
  price: number;
  description: string;
  type: string;
  rating: number;
  bedroomAmount: number;
  guestsAmount: number;
  items: string[];
  host: TestHost;
  reviews: TestReview[];
  coord: number[];
};

type TestOffers = TestOffer[];

export {TestOffer};
