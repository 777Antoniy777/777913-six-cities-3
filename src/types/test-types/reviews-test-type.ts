type TestReviewUser = {
  id: number;
  name: string;
  comment: string;
}

type TestReview = {
  id: number;
  user: TestReviewUser;
  status: string;
};

type TestReviews = TestReview[];

export {TestReviews};
