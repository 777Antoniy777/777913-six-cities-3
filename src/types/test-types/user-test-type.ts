type TestUser = {
  id: number;
  name: string;
  avatar: string;
  status: boolean;
};

export {TestUser};
