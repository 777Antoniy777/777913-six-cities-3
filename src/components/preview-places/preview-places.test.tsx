import * as React from "react";
import * as renderer from "react-test-renderer";
import PreviewPlaces from "./preview-places";
import {TestOffer} from "../../types/test-types/offers-test-type";
import {TestRouteLocation} from "../../types/test-types/location-test-type";

// set mocha data
const offer: TestOffer = {
  id: 1,
  city: {
    id: 1,
    name: `city`,
    coords: [1, 1],
  },
  title: `title 1`,
  premium: false,
  src: `img/image1`,
  photos: [`img/image1`],
  price: 999999,
  description: `test`,
  type: `type`,
  rating: 9999,
  bedroomAmount: 30,
  guestsAmount: 50,
  items: [`item`],
  host: {
    avatar: `img/avatar-1.jpg`,
    name: `name`,
    status: false,
  },
  reviews: [
    {
      id: 1,
      body: `text`,
      rating: 5,
      name: `name`,
      date: `date`,
    },
  ],
  coord: [1, 1],
};
const history = {};
const location: TestRouteLocation = {
  pathname: `/pathname`,
};

it(`render PreviewPlaces`, () => {
  const tree = renderer.create(
      <PreviewPlaces
        offer={offer}
        history={history}
        location={location}
      />)
      .toJSON();

  expect(tree).toMatchSnapshot();
});
