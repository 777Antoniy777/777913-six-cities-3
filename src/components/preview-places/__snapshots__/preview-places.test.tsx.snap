// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PreviewPlaces 1`] = `
<article
  className="place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="title 1"
        className="place-card__image"
        src="img/image1"
        style={
          Object {
            "height": 200,
            "width": 260,
          }
        }
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          999999
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="button place-card__bookmark-button"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height={19}
          width={18}
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "199980%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
      onClick={[Function]}
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        title 1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      type
    </p>
  </div>
</article>
`;
